satpy/readers/fci_l1c_fdhsi.py

```diff
@@ -125,61 +128,59 @@ def get_dataset(self, key, info=None):
         logger.debug('Reading {} from {}'.format(key.name, self.filename))
         # Get the dataset
         # Get metadata for given dataset
-        measured, root = self.get_channel_dataset(key.name)
-        radlab = measured + "/effective_radiance"
-        data = self[radlab]
+        measured = self.get_channel_group_path(key.name)
+        data = self[measured + "/effective_radiance"]
 
         attrs = data.attrs.copy()
         info = info.copy()
+
         fv = attrs.pop(
-                "FillValue",
-                default_fillvals.get(data.dtype.str[1:], np.nan))
-        vr = attrs.pop("valid_range", [-np.inf, np.inf])
+            "FillValue",
+            default_fillvals.get(data.dtype.str[1:], np.nan))
+        vr = attrs.get("valid_range", [-np.inf, np.inf])
         if key.calibration == "counts":
             attrs["_FillValue"] = fv
             nfv = fv
         else:
             nfv = np.nan
         data = data.where(data >= vr[0], nfv)
         data = data.where(data <= vr[1], nfv)
-        if key.calibration == "counts":
-            # from package description, this just means not applying add_offset
-            # and scale_factor
-            attrs.pop("scale_factor")
-            attrs.pop("add_offset")
-            data.attrs["units"] = "1"
-            res = data
-        else:
-            data = (data * attrs.pop("scale_factor", 1) +
-                    attrs.pop("add_offset", 0))
-
-            if key.calibration in ("brightness_temperature", "reflectance"):
-                res = self.calibrate(data, key, measured, root)
-            else:
-                res = data
-                data.attrs["units"] = attrs["units"]
+
+        res = self.calibrate(data, key)
+
         # pre-calibration units no longer apply
         info.pop("units")
         attrs.pop("units")
 
         res.attrs.update(key.to_dict())
         res.attrs.update(info)
         res.attrs.update(attrs)
+
         res.attrs["platform_name"] = self._platform_name_translate.get(
                 self["/attr/platform"], self["/attr/platform"])
+
+        # remove unpacking parameters for calibrated data
+        if key.calibration in ['brightness_temperature', 'reflectance']:
+            res.attrs.pop("add_offset")
+            res.attrs.pop("warm_add_offset")
+            res.attrs.pop("scale_factor")
+            res.attrs.pop("warm_scale_factor")
+
+        # remove attributes from original file which don't apply anymore
+        res.attrs.pop('long_name')
+
         return res
 
-    def get_channel_dataset(self, channel):
-        """Get channel dataset."""
-        root_group = 'data/{}'.format(channel)
-        group = 'data/{}/measured'.format(channel)
+    def get_channel_group_path(self, channel):
+        """Get channel group path."""
+        group_path = 'data/{}/measured'.format(channel)
```

Comment:
Ok, the name is better, but doesn't reflect what the function does :) I expect from the name to get the path to the channel group, but instead I get the path to `measured`. I don't have an opinion on what the function returns, just that the name of the function would match the output... Sorry for being so pedantic :)

Reply:
True! The old implementation that was delivering both group paths (channel and channel/measured), and I didn't update the name after removing the first.